diff --git a/lib/dns/server.js b/lib/dns/server.js
index b5237105..69be7c9b 100644
--- a/lib/dns/server.js
+++ b/lib/dns/server.js
@@ -447,7 +447,7 @@ class RootServer extends DNSServer {
     if (typeof this.middle === 'function') {
       let res;
       try {
-        res = await this.middle(tld, req);
+        res = await this.middle(tld, req, rinfo);
       } catch (e) {
         this.logger.warning(
           'Root server middleware resolution failed for name: %s',
